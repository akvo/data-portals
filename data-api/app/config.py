from typing import Any, Dict, Optional

from pydantic import BaseSettings, PostgresDsn, validator


class Settings(BaseSettings):
    POSTGRES_HOST: str
    POSTGRES_USER: str
    POSTGRES_PASSWORD: str
    POSTGRES_DB: str
    DATABASE_URI: Optional[PostgresDsn] = None

    @validator("DATABASE_URI", pre=True)
    def assemble_db_connection(cls, v: Optional[str], values: Dict[str, Any]) -> Any:
        if isinstance(v, str):  # pragma: no cover
            return v
        return PostgresDsn.build(
            scheme="postgresql",
            user=values.get("POSTGRES_USER"),
            password=values.get("POSTGRES_PASSWORD"),
            host=values.get("POSTGRES_HOST"),
            path=f"/{values.get('POSTGRES_DB') or ''}",
        )
