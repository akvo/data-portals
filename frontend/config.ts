export const DOMAIN = process.env.DOMAIN || ''

export const INTERNAL_API_DOMAIN = process.env.INTERNAL_API_DOMAIN || ''

export const API_PATH = process.env.API_PATH || ''

export const API_URL = DOMAIN + API_PATH

export const INTERNAL_API_URL = INTERNAL_API_DOMAIN + API_PATH

export const MAPBOX_TOKEN = process.env.MAPBOX_TOKEN || ''
