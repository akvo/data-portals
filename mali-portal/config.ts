export const API_HOST = process.env.API_HOST

export const DATA_ENDPOINT = process.env.DATA_ENDPOINT

export const DATA_API_ENDPOINT = process.env.DATA_API_ENDPOINT

export const DEFAULT_MAP_VIEWPORT = {
  latitude: 17.65,
  longitude: -4.15,
  zoom: 4.4,
}
