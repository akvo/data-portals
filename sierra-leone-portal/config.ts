export const API_HOST = process.env.API_HOST

export const DATA_ENDPOINT = process.env.DATA_ENDPOINT

export const DATA_API_ENDPOINT = process.env.DATA_API_ENDPOINT

export const DEFAULT_MAP_VIEWPORT = {
  latitude: 8.4,
  longitude: -11.8,
  zoom: 6.5,
}
