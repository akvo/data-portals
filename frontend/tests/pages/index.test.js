import React from 'react'
import { render } from '@testing-library/react'
import Home, { getServerSideProps, formatPercentage } from '../../pages/index'
import { server, rest } from '../mocks/server'
import { API_URL } from '../../config'

describe('Home page', () => {
  const fakeData = {
    Region: 'Foo',
    PMH: 1,
    Puits: 1,
    SAEP: 1,
  }

  it('matches snapshot', () => {
    const { asFragment } = render(<Home dataset={[fakeData]} />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call backend for dataset', async () => {
    server.use(
      rest.get(API_URL + '/functionality-rate-by-region', (_, res, ctx) => {
        return res(ctx.json([fakeData]))
      })
    )
    const response = await getServerSideProps()
    expect(response).toEqual({ props: { dataset: [fakeData] } })
  })
})

test('formatPercentage', () => {
  const result = formatPercentage(100)
  expect(result).toEqual('100%')
})
