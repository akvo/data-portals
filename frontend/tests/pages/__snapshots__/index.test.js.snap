// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Data portal WASH Mali
  </h1>
  <p>
    In this portal X, Y, Z. Please see the menu above for insights and data visualisations 
    <strong>
      per WASH theme
    </strong>
    , or go immediately to the 
    <strong>
      raw data
    </strong>
    :
  </p>
  <p>
    <button
      class="ant-btn ant-btn-primary"
      type="button"
    >
      <span>
        View raw data
      </span>
    </button>
  </p>
  <h2>
    Overview of data by region
  </h2>
  <p>
    The graph below gives an overview of the main indicators by region used by the DNH. Namely: functionality rate, access rate and equipment rate. For more information, see the data and graphics section.
  </p>
  <div
    style="height: 50vh; width: 100%; max-width: 1000px;"
  >
    <div
      style="width: 100%; height: 100%;"
    />
  </div>
</DocumentFragment>
`;
